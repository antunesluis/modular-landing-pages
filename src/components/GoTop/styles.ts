import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;

    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};

    border-radius: 0.4rem;
    transition: all 0.2s ease-in-out;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);

    &:hover {
      transform: translateY(-3px);
      box-shadow: 0 4px 8px rgba(0, 0, 0, 0.3);
      background: ${theme.colors.secondaryColor};
    }

    &:active {
      transform: translateY(0);
      box-shadow: 0 1px 2px rgba(0, 0, 0, 0.2);
      background: ${theme.colors.primaryColor};
    }

    > svg {
      width: 2.4rem;
      height: 2.4rem;
      transition: transform 0.2s ease-in-out;
    }

    &:hover > svg {
      transform: translateY(-3px);
    }

    &:active > svg {
      transform: translateY(0);
    }
  `}
`;
