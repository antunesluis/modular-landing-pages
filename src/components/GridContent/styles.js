import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.xlarge} 0 ${theme.spacings.xlarge};
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    width: 100%;
  `}
`;

export const HtmlContent = styled.div`
  ${({ theme }) => css`
    width: 100%;
    overflow-x: hidden; // Prevent horizontal scroll at component level

    p {
      font-size: ${theme.fonts.sizes.medium};
      line-height: 1.5;
      margin: ${theme.spacings.medium} 0;
    }

    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: ${theme.spacings.xlarge} 0 ${theme.spacings.xlarge};
      font-weight: 700;
      font-family: ${theme.fonts.family.secondary};
    }

    // Estilização para links
    a {
      color: ${theme.colors.secondaryColor};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }

    // Estilização para listas
    ul,
    ol {
      margin: ${theme.spacings.small} 0;
      padding: ${theme.spacings.medium} ${theme.spacings.large};
    }

    ul {
      list-style: disc;
    }

    ol {
      list-style: decimal;
    }

    li {
      text-align: left;
    }

    // Estilização para imagens
    img {
      max-width: 100%;
      height: auto;
    }

    // Estilização para citações
    blockquote {
      border-left: 0.5rem solid ${theme.colors.secondaryColor};
      color: ${theme.colors.primaryColor};
      margin: ${theme.spacings.medium} 0;
      padding: ${theme.spacings.medium} ${theme.spacings.large};
      font-style: italic;
    }

    // Estilização para tabelas
    table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
      display: block;
      overflow-x: auto; // Adiciona scroll horizontal apenas na tabela
      -webkit-overflow-scrolling: touch; // Melhora a rolagem em dispositivos touch

      // Remove display: block em telas maiores
      @media (min-width: 768px) {
        display: table;
      }
    }

    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 1px solid ${theme.colors.mediumGray};
      text-align: left;
      min-width: 15rem; // Garante uma largura mínima para células
    }

    table th {
      background: ${theme.colors.primaryColor};
      color: ${theme.colors.white};
    }
  `}
`;
