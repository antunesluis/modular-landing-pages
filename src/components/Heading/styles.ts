import styled, { css } from 'styled-components';
import { DefaultTheme } from 'styled-components';

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.xlarge};
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.xlarge};
  }
`;

const titleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

interface TitleProps {
  $colorDark?: boolean;
  size?: 'small' | 'medium' | 'big' | 'huge';
  $uppercase?: boolean;
}

export const Title = styled.h1<TitleProps>`
  ${({ theme, $colorDark, size = 'huge', $uppercase = false }) => css`
    color: ${$colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase($uppercase)};
  `}
`;
