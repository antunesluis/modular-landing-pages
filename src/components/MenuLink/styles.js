import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    position: relative;
    font-size: ${theme.fonts.sizes.small};
    padding: ${theme.spacings.small};
    color: ${theme.colors.primaryColor};

    &::after {
      content: '';
      position: absolute;
      bottom: 0.8rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.secondaryColor};
      transition: all 300ms ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
