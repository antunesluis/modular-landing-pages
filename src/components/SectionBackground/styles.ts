import styled, { css } from 'styled-components';
import { DefaultTheme } from 'styled-components';

const containerBackgroundActivate = (theme: DefaultTheme) => css`
  background-image: url(${theme.backgroundImage});
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
  color: ${theme.colors.white};
`;

interface ContainerProps {
  $background?: boolean;
}

export const Container = styled.div<ContainerProps>`
  ${({ theme, $background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};

    ${$background && containerBackgroundActivate(theme)};

    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
