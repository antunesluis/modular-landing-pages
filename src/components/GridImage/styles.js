import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.spacings.maxWidth};
    margin: 0 auto;
    position: relative;

    > ${TextComponent} {
      opacity: 0.9;
      max-width: 100%;
    }

    @media ${theme.media.lteMedium} {
      padding: ${theme.spacings.large} ${theme.spacings.small};

      > ${TextComponent} {
        margin: ${theme.spacings.medium} 0 ${theme.spacings.xhuge};
      }
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    width: 100%;
    display: grid;
    grid-template-columns: repeat(3, 1fr);

    gap: ${theme.spacings.large};
    margin: ${theme.spacings.large} auto;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.medium};
      margin: ${theme.spacings.medium} auto;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    aspect-ratio: 1; // Mantém proporção quadrada
    overflow: hidden;
    position: relative;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);

    border-radius: ${theme.spacings.small};
    background: ${theme.colors.mediumGray};

    &::before {
      content: '';
      z-index: 1;
      opacity: 0;
      position: absolute;
      inset: 0; // Substitui top: 0, left: 0, width: 100%, height: 100%

      background: rgba(0, 0, 0, 0.3);
      transition: opacity 300ms ease-in-out;
    }

    &:hover {
      &::before {
        opacity: 1;
      }

      img {
        transform: scale(1.1) rotate(3deg);
      }
    }
  `}
`;

export const Image = styled.img`
  ${() => css`
    display: block; // Remove espaço extra abaixo da imagem
    width: 100%;
    height: 100%;
    object-fit: cover;
    vertical-align: middle;
    transition: transform 500ms ease-in-out;
    will-change: transform;
  `}
`;

export const ImageCaption = styled.figcaption`
  ${({ theme }) => css`
    position: absolute;
    opacity: 0;
    bottom: 0;
    right: 0; // Garante que ocupe toda a largura
    left: 0;
    width: 100%;
    z-index: 2;

    background: linear-gradient(to top, rgba(0, 0, 0, 0.8), transparent);
    padding: ${theme.spacings.small}; // Adicionado padding
    color: ${theme.colors.white};
    font-size: ${theme.fonts.sizes.small};

    transform: translateY(100%);
    transition: all 300ms ease-in-out;

    ${GridElement}:hover & {
      opacity: 1;
      transform: translateY(0);
    }
  `}
`;
