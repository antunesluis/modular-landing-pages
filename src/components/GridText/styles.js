import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.spacings.maxWidth};
    margin: 0 auto;

    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
      opacity: 0.9;
    }

    > ${HeadingContainer} {
      text-shadow: 0.1rem 0.1rem 0.2rem rgba(0, 0, 0, 0.2);
      margin-bottom: ${theme.spacings.medium};
    }

    @media ${theme.media.lteMedium} {
      ${TextComponent} {
        margin-bottom: ${theme.spacings.large};
      }
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    max-width: 100%;

    grid-template-columns: repeat(auto-fit, minmax(min(100%, 32rem), 1fr));
    gap: ${theme.spacings.xlarge};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    position: relative;
    overflow: hidden;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
    transition: all 0.3s ease-in-out;
    border-radius: 10px;

    padding: ${theme.spacings.medium};
    background: rgba(255, 255, 255, 0.03);
    border-radius: ${theme.spacings.tiny};

    &:hover {
      transform: translateY(-6px);
      background: rgba(255, 255, 255, 0.05);
    }

    ${TextComponent} {
      opacity: 0.85;
      line-height: 1.6;
    }
  `}
`;

export const HeaderContainer = styled.div`
  ${() => css`
    ${({ theme }) => css`
      display: flex;
      align-items: center;
      margin-bottom: ${theme.spacings.medium};
      position: relative;

      ${HeadingContainer} {
        margin: 0;
        padding-left: ${theme.spacings.medium};
      }
    `}
  `}
`;

export const CardNumber = styled.span`
  ${({ theme }) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    font-weight: 700;
    // font-weight: bold;
    // font-size: clamp(5rem, 8vw, 7.5rem);

    opacity: 0.15;
    transition: all 0.3s ease-in-out;

    ${GridElement}:hover & {
      opacity: 0.2;
      transform: scale(1.1) rotate(5deg);
    }
  `}
`;
