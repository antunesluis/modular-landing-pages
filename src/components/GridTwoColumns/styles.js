import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: minmax(280px, 1.2fr) 1.3fr;
    align-items: center;
    gap: ${theme.spacings.xlarge};

    margin: 0 auto;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
      gap: ${theme.spacings.small};
    }

    ${Title} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    order: 1;
    max-width: 60ch;

    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacings.medium};
      order: 2;
      max-width: 100%;
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    order: 2;
    width: 100%;
    margin: 0 auto;
    padding: ${theme.spacings.small};

    @media ${theme.media.lteLarge} {
      max-width: 600px;
    }

    @media ${theme.media.lteMedium} {
      order: 1;
      margin-bottom: ${theme.spacings.medium};
      max-width: 500px;
    }
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    height: auto;
    object-fit: cover;
    transition: all 300ms ease-in-out;

    @media ${theme.media.lteMedium} {
      max-height: 40rem;
    }
  `}
`;
