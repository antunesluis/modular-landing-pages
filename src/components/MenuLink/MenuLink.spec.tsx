import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { MenuLink } from '.';

vi.mock('next/link', () => ({
  default: ({
    children,
    href,
  }: {
    children: React.ReactNode;
    href: string;
  }) => <a href={href}>{children}</a>,
}));

describe('<MenuLink />', () => {
  it('should render a Next.js Link for internal links', () => {
    renderTheme(<MenuLink link="/internal-link">Internal</MenuLink>);
    const link = screen.getByRole('link', { name: 'Internal' });
    expect(link).toHaveAttribute('href', '/internal-link');
  });

  it('should render a normal anchor tag for external links', () => {
    renderTheme(<MenuLink link="http://external-link.com">External</MenuLink>);
    const link = screen.getByRole('link', { name: 'External' });
    expect(link).toHaveAttribute('href', 'http://external-link.com');
    expect(link.tagName).toBe('A');
  });

  it('should render with a target _self by default', () => {
    renderTheme(<MenuLink link="http://localhost">Children</MenuLink>);
    const link = screen.getByRole('link', { name: 'Children' });
    expect(link).toHaveAttribute('target', '_self');
  });

  it('should render open in a new tab', () => {
    renderTheme(
      <MenuLink link="http://localhost" newTab={true}>
        Children
      </MenuLink>,
    );
    expect(screen.getByRole('link', { name: 'Children' })).toHaveAttribute(
      'target',
      '_blank',
    );
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <MenuLink link="http://localhost" newTab={false}>
        Children
      </MenuLink>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
