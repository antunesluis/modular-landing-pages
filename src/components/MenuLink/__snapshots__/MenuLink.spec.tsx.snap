// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MenuLink /> > should match snapshot 1`] = `
.c0 {
  display: block;
  text-decoration: none;
  position: relative;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.8rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

<a
  class="c0"
  href="http://localhost"
  target="_self"
>
  Children
</a>
`;
